### PR TITLE
Add restore option

### PR DESCRIPTION
**Pull request by yinzara**
Some tests may want to use the natural system environment but then restore it completely after the test (so that any environment variables added during the test get removed).  Adding a 'restore' option allows for this.

### CODE REVIEW

**Comment by bahmutov** [comment]
I like this option, but would like to see a test added

**Comment by yinzara** [comment]
Tests added. Do you want me to rebase the branch as there are commits that don't pass lint.

**Comment by yinzara** [comment]
I also took the liberty of adding TypeScript definitions.

File: src/mocked-env-spec.js [hunk 1]
```diff
@@ -189,4 +203,68 @@ describe('mocked-env', () => {
...
+      restore = mockedEnv(
+        {
+          FOO: 'foo',
+          BAR: 'bar'
+        },
+        { restore: true }
+      )
+    })
+
+    it('has default env, BANG, FOO and BAR', () => {
```
Comment (bahmutov) [inline src/mocked-env-spec.js]: hmm, I am not sure this is the best test for this. We want to check "restore" right? So we probably better do
```
const restore = mockedEnv({...}, {restore: true})
process.env.BANG = 'bash'
restore()
la(process.env.BANG === undefined, 'did not restore process.env')
```

what do you think?

Comment (yinzara) [reply src/mocked-env-spec.js]: That test only verifies that the environment variable that was present before mocking the environment was available during the mock when "restore" is given.

The later test "pass 'restore' in options" actually tests the case you're referring to where it changes the variable via process.env after mocking then verifies it restores it after.

Comment (bahmutov) [reply src/mocked-env-spec.js]: I see what you mean, merging your PR now

**Pull request merged**